src/BookmarkWindow.cpp

```diff
@@ -24,26 +24,23 @@ static const char* kCancel = B_TRANSLATE("Cancel");
 static const char* kOK = B_TRANSLATE("OK");
 
 TBookmarkWindow::TBookmarkWindow(float x, float y, const char *title, const char *dir)
-	:	BWindow(BRect(x, y, x + 580, y + 500), title, B_FLOATING_WINDOW_LOOK,
-				B_MODAL_APP_WINDOW_FEEL, B_NOT_RESIZABLE | B_NOT_ZOOMABLE)
+	:	BWindow(BRect(x, y, 0,0), title, B_FLOATING_WINDOW_LOOK,
+				B_MODAL_APP_WINDOW_FEEL, B_NOT_RESIZABLE | B_NOT_ZOOMABLE | B_AUTO_UPDATE_SIZE_LIMITS )
 {
 	fBookmarkDir.SetTo(dir);
-	fNewBookmark = true;
 	
-	BView *bgview = new BView(Bounds(), "bgview", B_FOLLOW_NONE, 0);
-	bgview->SetViewColor(217, 217, 217);
-	AddChild(bgview);
+	fNewBookmark = true;
 	
 	float lx = 10, rx = Bounds().right - 10, div = 70;
 	
-	fBookmarkName = new TOAEdit(BRect(lx, 20, rx, 38), "BookmarkName", B_TRANSLATE("Name:"), B_TRANSLATE("New bookmark"), new BMessage(EDIT_CHANGED));
-	fHostAddress = new TOAEdit(BRect(lx, 40, rx, 58), "HostAddress", B_TRANSLATE("Address:"), "", new BMessage(EDIT_CHANGED));
-	fPort = new TOAEdit(BRect(lx, 60, rx, 78), "Port", B_TRANSLATE("Port:"), "21", new BMessage(EDIT_CHANGED));
-	fUsername = new TOAEdit(BRect(lx, 80, rx, 98), "Username", B_TRANSLATE("User name:"), "", new BMessage(EDIT_CHANGED));
-	fPassword = new TOAEdit(BRect(lx, 100, rx, 118), "Password", B_TRANSLATE("Password:"), "", new BMessage(EDIT_CHANGED));
-	fEncoder = new TOAEdit(BRect(lx, 120, rx, 138), "Encoder", B_TRANSLATE("Encoder:"), "", NULL);
-	fRemotePath = new TOAEdit(BRect(lx, 140, rx, 158), "RemotePath", B_TRANSLATE("Remote path:"), "", NULL);
-	fLocalPath = new TOAEdit(BRect(lx, 160, rx, 178), "LocalPath", B_TRANSLATE("Local path:"), TFtpPositive::GetDefaultLocalDir().String(), NULL);
+	fBookmarkName = new TOAEdit(BRect(0,0,1,1),"BookmarkName", B_TRANSLATE("Name:"), B_TRANSLATE("New bookmark"), new BMessage(EDIT_CHANGED));
```

Comment:
When using layout managment, you need to use the layout-aware constructors, i.e. without the BRect for the frame etc. See https://www.haiku-os.org/docs/api/classBTextControl.html
You'll have to adjust OAEdit.cpp